internal/workerutil/dbworker/store/store.go

```diff
@@ -300,115 +298,85 @@ SELECT COUNT(*) FROM %s WHERE (
 ) %s
 `
 
-// Dequeue selects the first unlocked record matching the given conditions and locks it in a new transaction that
-// should be held by the worker process. If there is such a record, it is returned along with a new store instance
-// that wraps the transaction. The resulting transaction must be closed by the caller, and the transaction should
-// include a state transition of the record into a terminal state. If there is no such unlocked record, a nil record
-// and a nil store will be returned along with a false-valued flag. This method must not be called from within a
-// transaction.
+// Dequeue selects the first queued record matching the given conditions and updates the state to processing. If there
+// is such a record, it is returned. If there is no such unclaimed record, a nil record and and a nil cancel function
+// will be returned along with a false-valued flag. This method must not be called from within a transaction.
+//
+// A background goroutine that continuously updates the record's last modified time will be started. The returned cancel
+// function should be called once the record no longer needs to be locked from selection or reset by another process.
+// Most often, this will be when the handler moves the record into a terminal state.
 //
 // The supplied conditions may use the alias provided in `ViewName`, if one was supplied.
-func (s *store) Dequeue(ctx context.Context, workerHostname string, conditions []*sqlf.Query) (record workerutil.Record, _ Store, exists bool, err error) {
-	return s.dequeue(ctx, workerHostname, conditions, false)
-}
-
-// DequeueWithIndependentTransactionContext is like Dequeue, but will use a context.Background() for the underlying
-// transaction context. This method allows the transaction to lexically outlive the code in which it was created. This
-// is useful if a longer-running transaction is managed explicitly between multiple goroutines.
-func (s *store) DequeueWithIndependentTransactionContext(ctx context.Context, workerHostname string, conditions []*sqlf.Query) (workerutil.Record, Store, bool, error) {
-	return s.dequeue(ctx, workerHostname, conditions, true)
-}
-
-func (s *store) dequeue(ctx context.Context, workerHostname string, conditions []*sqlf.Query, independentTxCtx bool) (record workerutil.Record, _ Store, exists bool, err error) {
+func (s *store) Dequeue(ctx context.Context, workerHostname string, conditions []*sqlf.Query) (_ workerutil.Record, _ context.CancelFunc, _ bool, err error) {
 	ctx, traceLog, endObservation := s.operations.dequeue.WithAndLogger(ctx, &err, observation.Args{})
 	defer endObservation(1, observation.Args{})
 
 	if s.InTransaction() {
 		return nil, nil, false, ErrDequeueTransaction
 	}
 
-	txCtx := ctx
-	if independentTxCtx {
-		txCtx = context.Background()
-	}
+	now := s.options.clock.Now()
```

Comment:
Based on https://github.com/sourcegraph/sourcegraph/pull/20936/files#r632445026 it sounds like this needs a `UTC()`?

```suggestion
	now := s.options.clock.Now().UTC()
```

Reply:
I think it would be safe to put UTC on all now calls in this file. Times are stored with timezones so as long as it's consistent it will help us with testing/finding patterns.

Reply:
Updated.